flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/operations/DescribeFunctionOperation.java

```diff
@@ -0,0 +1,125 @@
+
+import static org.apache.flink.table.api.internal.TableResultUtils.buildTableResult;
+import static org.apache.flink.table.types.inference.TypeInferenceUtil.generateSignature;
+
+/**
+ * Operation to describe a DESCRIBE [EXTENDED] [[catalogName.] dataBasesName].sqlIdentifier
+ * statement.
+ */
+@Internal
+public class DescribeFunctionOperation implements Operation, ExecutableOperation {
+
+    private final UnresolvedIdentifier sqlIdentifier;
+    private final boolean isExtended;
+
+    public DescribeFunctionOperation(UnresolvedIdentifier sqlIdentifier, boolean isExtended) {
+        this.sqlIdentifier = sqlIdentifier;
+        this.isExtended = isExtended;
+    }
+
+    public UnresolvedIdentifier getSqlIdentifier() {
+        return sqlIdentifier;
+    }
+
+    public boolean isExtended() {
+        return isExtended;
+    }
+
+    @Override
+    public String asSummaryString() {
+        Map<String, Object> params = new LinkedHashMap<>();
+        params.put("identifier", sqlIdentifier);
+        params.put("isExtended", isExtended);
+        return OperationUtils.formatWithChildren(
+                "DESCRIBE FUNCTION", params, Collections.emptyList(), Operation::asSummaryString);
+    }
+
+    @Override
+    public TableResultInternal execute(Context ctx) {
+        // DESCRIBE FUNCTION <function> shows all the function properties.
+        Optional<ContextResolvedFunction> functionOpt =
+                ctx.getFunctionCatalog().lookupFunction(sqlIdentifier);
+        if (!functionOpt.isPresent()) {
+            throw new ValidationException(
+                    String.format(
+                            "Function with the identifier '%s' doesn't exist.",
+                            sqlIdentifier.asSummaryString()));
+        }
+        final ContextResolvedFunction function = functionOpt.get();
+        final CatalogFunction catalogFunction = function.getCatalogFunction();
+
+        List<List<Object>> rows = new ArrayList<>();
+        rows.add(Arrays.asList("system function", String.valueOf(catalogFunction == null)));
+        rows.add(Arrays.asList("temporary", String.valueOf(function.isTemporary())));
+        if (catalogFunction != null) {
+            rows.add(Arrays.asList("class name", catalogFunction.getClassName()));
+            rows.add(
+                    Arrays.asList(
+                            "function language", catalogFunction.getFunctionLanguage().toString()));
+            rows.add(
+                    Arrays.asList(
+                            "resource uris", catalogFunction.getFunctionResources().toString()));
+        }
+
+        if (isExtended) {
+            final FunctionDefinition definition = function.getDefinition();
+            rows.add(Arrays.asList("kind", definition.getKind().toString()));
+            rows.add(Arrays.asList("requirements", definition.getRequirements().toString()));
+            rows.add(Arrays.asList("deterministic", String.valueOf(definition.isDeterministic())));
```

Comment:
can we use `is determinstic`, `is temporary` and `supports constant folding`, `is system function`? reads nicer.

Reply:
added